README.md
# Convertor
Un simplu convertor


Changelog.txt
version 0.1(alpha)
-added title
version 0.2(alpha)
-added background,margins and a button that does nothing
-added easteregg(Come find me!!!)
version 0.3(alpha)
-finished the main menu
-buttons change the font color when you hover the cursor over them
-different texture rendering for different fonts
version 0.4(beta)
-beta,yuhuu
-added submenu
-added all kinds of dynamic buttons with nice rendering(butonas is the best)
-"lungime" submenu is done,10 more to go
-added input function
version 0.5(beta)
-fixed some graphical issues
-Happy Christmas(25.12.2016)
version 0.(6)(beta)
-still beta :(
-added all the menus(it works!!)
-kinda ready
version 1.0(full release)
-fixed some graphical issues(part 2)
-added music
-added an icon
-did some organization
-added a nice pic and a title for each submenu


main.cpp
#include <iostream>
#include <fstream>
using namespace std;
ifstream fin1("Convert.in");
#include <stdio.h>
#define _WIN32_WINNT 0x0500
#include <windows.h>
#include <conio.h>
HWND hConsole;

MENUITEMINFO mii;
void centerconsole()
{
    	//Get the window console handle(isn't the right code but works for these sample
	HWND ConsoleWindow;
	ConsoleWindow=GetForegroundWindow();

	//Getting the desktop hadle and rectangule
	HWND   hwndScreen;
	RECT   rectScreen;
	hwndScreen=GetDesktopWindow ();
	GetWindowRect(hwndScreen,&rectScreen);



	// Get the current width and height of the console
	RECT rConsole;
	GetWindowRect (ConsoleWindow,&rConsole);
	int Width = rConsole.left = rConsole.right;
	int Height = rConsole.bottom - rConsole.top;

	//caculate the window console to center of the screen
	int ConsolePosX;
	int ConsolePosY;
	ConsolePosX = ((rectScreen.right-rectScreen.left)/2-Width/2 );
	ConsolePosY = ((rectScreen.bottom-rectScreen.top)/2- Height/2 );
	SetWindowPos(ConsoleWindow,NULL,ConsolePosX,ConsolePosY, Width, Height, SWP_SHOWWINDOW || SWP_NOSIZE);

	_getch();
}
void misc()
{


    SMALL_RECT windowSize = {0, 0, 100, 100};

    SetConsoleWindowInfo(hConsole, TRUE, &windowSize);

    SetConsoleTitle("CONVERTOR");

    Sleep(10); //Let the window to update the Title!

    hConsole=FindWindow(NULL, "CONVERTOR");

    HMENU hMenu = GetSystemMenu ( (HWND)hConsole, FALSE);



    mii.fState = MFS_GRAYED;

    EnableMenuItem(hMenu,4,MF_DISABLED);

    SetMenuItemInfo(hMenu,4,MF_BYPOSITION,&mii);



    DeleteMenu( hMenu, 1, MF_BYPOSITION); // disable 'Move'

    DeleteMenu( hMenu, 1, MF_BYPOSITION); // disable 'Size'


    DeleteMenu( hMenu, 2, MF_BYPOSITION); // disable 'Maximize'

    HWND consoleWindow  = GetConsoleWindow();//1.
    LONG style = GetWindowLong(consoleWindow, GWL_STYLE); //2.
    style = style & ~(WS_MINIMIZEBOX);//3.
    SetWindowLong(consoleWindow, GWL_STYLE, style);//4.disable 'minimize'

    //DeleteMenu( hMenu, 3, MF_BYPOSITION); // disable 'Close or X'


}
void SetConsoleWindowSize(int x, int y)
{
    HANDLE h = GetStdHandle(STD_OUTPUT_HANDLE);

    // If either dimension is greater than the largest console window we can have,
    // there is no point in attempting the change.
  COORD largestSize = GetLargestConsoleWindowSize(h);


    CONSOLE_SCREEN_BUFFER_INFO bufferInfo;
GetConsoleScreenBufferInfo(h, &bufferInfo);


    SMALL_RECT& winInfo = bufferInfo.srWindow;
    COORD windowSize = { winInfo.Right - winInfo.Left + 1, winInfo.Bottom - winInfo.Top + 1};

    if (windowSize.X > x || windowSize.Y > y)
    {
        // window size needs to be adjusted before the buffer size can be reduced.
        SMALL_RECT info =
        {
            0,
            0,
            x < windowSize.X ? x-1 : windowSize.X-1,
            y < windowSize.Y ? y-1 : windowSize.Y-1
        };

        SetConsoleWindowInfo(h, TRUE, &info);

    }

    COORD size = { 101,y+5 };
    SetConsoleScreenBufferSize(h, size);



    SMALL_RECT info = { 0, 0, x - 1, y - 1 };
    SetConsoleWindowInfo(h, TRUE, &info);

}

void printmargins(int width,int height)
{
    SetConsoleTextAttribute (GetStdHandle(STD_OUTPUT_HANDLE), FOREGROUND_RED |
                             FOREGROUND_GREEN |
                             FOREGROUND_INTENSITY |
                             BACKGROUND_BLUE
                            );
    cout<<(char)201;
    int i,j;
    for(i=2; i<width; i++)
        cout<<(char)205;
    cout<<(char)187;
    cout<<endl;
    cout<<(char)186<<(char)201;
    for(i=3; i<width-1; i++)
        cout<<(char)205;
    cout<<(char)187<<(char)186<<endl;
    for(i=3; i<height-1; i++)
    {
        cout<<(char)186<<(char)186;
        for(j=3; j<width-1; j++)
            cout<<" ";
        cout<<(char)186<<(char)186;
        cout<<endl;
    }
    cout<<(char)186<<(char)200;
    for(i=3; i<width-1; i++)
        cout<<(char)205;
    cout<<(char)188<<(char)186;
    cout<<endl;
    cout<<(char)200;
    for(i=2; i<width; i++)
        cout<<(char)205;
    cout<<(char)188;
}
void gotoXY(int x, int y)
{
    //Initialize the coordinates
    COORD coord = {x, y};
    //Set the position
    SetConsoleCursorPosition(GetStdHandle(STD_OUTPUT_HANDLE), coord);
    return;
}
void start()
{
    SetConsoleWindowSize(100,50);
    misc();

    printmargins(100,50);
    int i,j;
    char x;
    for(i=0; i<8; i++)
    {
        for(j=0; j<63; j++)
        {
            gotoXY(19+j,5+i);
            fin1>> noskipws >>x;
            cout<<(char)x;
        }
        cout<<endl;
    }
centerconsole();
}
int main()
{
    start();
    gotoXY(0,51);
    /*HWND hWnd = GetConsoleWindow();
    RECT rct;
    int width, height;
    if (GetWindowRect(hWnd, &rct))
    {
        width = rct.right - rct.left;
        height = rct.bottom - rct.top;
    }
    cout<<width<<" by "<<height<<endl;*/

    Sleep(6000);
}


render.h
#include <iostream>
#include <SDL.h>
#include <SDL_image.h>
#include <stdio.h>
#include <string>
#include <cmath>
#include <SDL_ttf.h>


//The window renderer
SDL_Renderer* gRenderer = NULL;

//Globally used font
TTF_Font *gFont = NULL;
TTF_Font *gFont2 = NULL;
TTF_Font *gFont3 = NULL;
TTF_Font *gFont4 = NULL;
TTF_Font *gFont5 = NULL;



class LTexture
{
public:
    //Initializes variables
    LTexture();

    //Deallocates memory
    ~LTexture();

    //Loads image at specified path
    bool loadFromFile( std::string path );

    //Creates image from font string
    bool loadFromRenderedText( std::string textureText, SDL_Color textColor,TTF_Font *gFont);

    //Deallocates texture
    void free();

    //Set color modulation
    void setColor( Uint8 red, Uint8 green, Uint8 blue );

    //Set blending
    void setBlendMode( SDL_BlendMode blending );

    //Set alpha modulation
    void setAlpha( Uint8 alpha );

    //Renders texture at given point
    void render( int x, int y, SDL_Rect* clip = NULL, double angle = 0.0, SDL_Point* center = NULL, SDL_RendererFlip flip = SDL_FLIP_NONE );

    //Gets image dimensions
    int getWidth();
    int getHeight();

private:
    //The actual hardware texture
    SDL_Texture* mTexture;

    //Image dimensions
    int mWidth;
    int mHeight;
};






LTexture::LTexture()
{
    //Initialize
    mTexture = NULL;
    mWidth = 0;
    mHeight = 0;
}

LTexture::~LTexture()
{
    //Deallocate
    free();
}

bool LTexture::loadFromFile( std::string path )
{
    //Get rid of preexisting texture
    free();

    //The final texture
    SDL_Texture* newTexture = NULL;

    //Load image at specified path
    SDL_Surface* loadedSurface = IMG_Load( path.c_str() );
    if( loadedSurface == NULL )
    {
        printf( "Unable to load image %s! SDL_image Error: %s\n", path.c_str(), IMG_GetError() );
    }
    else
    {
        //Color key image
        SDL_SetColorKey( loadedSurface, SDL_TRUE, SDL_MapRGB( loadedSurface->format, 0, 0xFF, 0xFF ) );

        //Create texture from surface pixels
        newTexture = SDL_CreateTextureFromSurface( gRenderer, loadedSurface );
        if( newTexture == NULL )
        {
            printf( "Unable to create texture from %s! SDL Error: %s\n", path.c_str(), SDL_GetError() );
        }
        else
        {
            //Get image dimensions
            mWidth = loadedSurface->w;
            mHeight = loadedSurface->h;
        }

        //Get rid of old loaded surface
        SDL_FreeSurface( loadedSurface );
    }

    //Return success
    mTexture = newTexture;
    return mTexture != NULL;
}

bool LTexture::loadFromRenderedText( std::string textureText, SDL_Color textColor,TTF_Font *gFont )
{
    //Get rid of preexisting texture
    free();

    //Render text surface
    SDL_Surface* textSurface = TTF_RenderText_Solid( gFont, textureText.c_str(), textColor );
    if( textSurface == NULL )
    {
        printf( "Unable to render text surface! SDL_ttf Error: %s\n", TTF_GetError() );
    }
    else
    {
        //Create texture from surface pixels
        mTexture = SDL_CreateTextureFromSurface( gRenderer, textSurface );
        if( mTexture == NULL )
        {
            printf( "Unable to create texture from rendered text! SDL Error: %s\n", SDL_GetError() );
        }
        else
        {
            //Get image dimensions
            mWidth = textSurface->w;
            mHeight = textSurface->h;
        }

        //Get rid of old surface
        SDL_FreeSurface( textSurface );
    }

    //Return success
    return mTexture != NULL;
}

void LTexture::free()
{
    //Free texture if it exists
    if( mTexture != NULL )
    {
        SDL_DestroyTexture( mTexture );
        mTexture = NULL;
        mWidth = 0;
        mHeight = 0;
    }
}

void LTexture::setColor( Uint8 red, Uint8 green, Uint8 blue )
{
    //Modulate texture rgb
    SDL_SetTextureColorMod( mTexture, red, green, blue );
}

void LTexture::setBlendMode( SDL_BlendMode blending )
{
    //Set blending function
    SDL_SetTextureBlendMode( mTexture, blending );
}

void LTexture::setAlpha( Uint8 alpha )
{
    //Modulate texture alpha
    SDL_SetTextureAlphaMod( mTexture, alpha );
}

void LTexture::render( int x, int y, SDL_Rect* clip, double angle, SDL_Point* center, SDL_RendererFlip flip )
{
    //Set rendering space and render to screen
    SDL_Rect renderQuad = { x, y, mWidth, mHeight };

    //Set clip rendering dimensions
    if( clip != NULL )
    {
        renderQuad.w = clip->w;
        renderQuad.h = clip->h;
    }

    //Render to screen
    SDL_RenderCopyEx( gRenderer, mTexture, clip, &renderQuad, angle, center, flip );
}

int LTexture::getWidth()
{
    return mWidth;
}

int LTexture::getHeight()
{
    return mHeight;
}


SDL_Texture* loadTexture( std::string path )
{
    //The final texture
    SDL_Texture* newTexture = NULL;

    //Load image at specified path
    SDL_Surface* loadedSurface = IMG_Load( path.c_str() );
    if( loadedSurface == NULL )
    {
        printf( "Unable to load image %s! SDL_image Error: %s\n", path.c_str(), IMG_GetError() );
    }
    else
    {
        //Create texture from surface pixels
        newTexture = SDL_CreateTextureFromSurface( gRenderer, loadedSurface );
        if( newTexture == NULL )
        {
            printf( "Unable to create texture from %s! SDL Error: %s\n", path.c_str(), SDL_GetError() );
        }

        //Get rid of old loaded surface
        SDL_FreeSurface( loadedSurface );
    }

    return newTexture;
}



files+matrixes.h


ifstream fin1("lungime.txt");
ifstream fin2("arie.txt");
ifstream fin3("volum.txt");
ifstream fin4("timp.txt");
ifstream fin5("viteza.txt");
ifstream fin6("temperatura.txt");
ifstream fin7("masa.txt");
ifstream fin8("energie.txt");
ifstream fin9("presiune.txt");
ifstream fin10("densitate.txt");
ifstream fin11("consum.txt");

char lungime[12][12];
char arie[12][12];
char volum[13][12];
char timp[12][12];
char viteza[23][12];
char temperatura[4][12];
char masa[9][12];
char energie[8][16];
char presiune[19][12];
char densitate[21][17];
char consum[7][12];



void citire_lungime(char a[12][12])
{
    char x[12];
    int i=0,j;
    while(fin1>>x)
    {
        j=0;
        while(x[j]!=NULL)
        {
            a[i][j]=x[j];
            j++;
        }
        i++;
    }
}

void citire_arie(char a[12][12])
{
    char x[12];
    int i=0,j;
    while(fin2>>x)
    {
        j=0;
        while(x[j]!=NULL)
        {
            a[i][j]=x[j];
            j++;
        }
        i++;
    }
}

void citire_volum(char a[13][12])
{
    char x[12];
    int i=0,j;
    while(fin3>>x)
    {
        j=0;
        while(x[j]!=NULL)
        {
            a[i][j]=x[j];
            j++;
        }
        i++;
    }
}

void citire_timp(char a[12][12])
{
    char x[12];
    int i=0,j;
    while(fin4>>x)
    {
        j=0;
        while(x[j]!=NULL)
        {
            a[i][j]=x[j];
            j++;
        }
        i++;
    }
}
void citire_viteza(char a[23][12])
{
    char x[12];
    int i=0,j;
    while(fin5>>x)
    {
        j=0;
        while(x[j]!=NULL)
        {
            a[i][j]=x[j];
            j++;
        }
        i++;
    }
}

void citire_temperatura(char a[4][12])
{
    char x[12];
    int i=0,j;
    while(fin6>>x)
    {
        j=0;
        while(x[j]!=NULL)
        {
            a[i][j]=x[j];
            j++;
        }
        i++;
    }
}

void citire_masa(char a[9][12])
{
    char x[12];
    int i=0,j;
    while(fin7>>x)
    {
        j=0;
        while(x[j]!=NULL)
        {
            a[i][j]=x[j];
            j++;
        }
        i++;
    }
}


void citire_energie(char a[8][16])
{
    char x[16];
    int i=0,j;
    while(fin8>>x)
    {
        j=0;
        while(x[j]!=NULL)
        {
            a[i][j]=x[j];
            j++;
        }
        i++;
    }
}

void citire_presiune(char a[19][12])
{
    char x[12];
    int i=0,j;
    while(fin9>>x)
    {
        j=0;
        while(x[j]!=NULL)
        {
            a[i][j]=x[j];
            j++;
        }
        i++;
    }
}

void citire_densitate(char a[21][17])
{
    char x[17];
    int i=0,j;
    while(fin10>>x)
    {
        j=0;
        while(x[j]!=NULL)
        {
            a[i][j]=x[j];
            j++;
        }
        i++;
    }
}

void citire_consum(char a[7][12])
{
    char x[12];
    int i=0,j;
    while(fin11>>x)
    {
        j=0;
        while(x[j]!=NULL)
        {
            a[i][j]=x[j];
            j++;
        }
        i++;
    }
}


transformari.h
#include <cstring>
#include <iomanip>
#include <math.h>
#include <stdio.h>


long double trans_char_float(char x[])
{
    char *p = x, *q = strchr(x, '.');
    if(x[0]=='-')
        p++;
    long double a = 0, k = 0.1, b = 0;
    while(p != q && *p != '\0')
    {
        a = a* 10 + *p - '0';
        p++;
    }

    if(!q)
    {
        if(x[0]=='-')
            return -a;
        return a;
    }

    ++q;
    while(*q != '\0')
    {
        b = *q - '0';
        a = a + b * k;
        k /= 10;
        ++q;
    }
    if(x[0]=='-')
        return -a;
    return a;

}

char trans_lungime(char x[256],char y[256],int i,int j)
{
    long double f=trans_char_float(x);

    switch (i)
    {
    case 0 :
        f=0.01*f;
        break;
    case 1 :
        f=0.3048*f;
        break;
    case 2 :
        f=0.0254*f;
        break;
    case 3 :
        f=1000*f;
        break;
    case 4 :
        f=1*f;
        break;
    case 5 :
        f=1609.344*f;
        break;
    case 6 :
        f=0.001*f;
        break;
    case 7 :
        f=1852*f;
        break;
    case 8 :
        f=0.9144002494*f;
        break;
    }

    switch (j)
    {
    case 0 :
        f=100*f;
        break;
    case 1 :
        f=3.280839895013123*f;
        break;
    case 2 :
        f=39.37007874015748*f;
        break;
    case 3 :
        f=0.001*f;
        break;
    case 4 :
        f=1*f;
        break;
    case 5 :
        f=0.0006213711922373339*f;
        break;
    case 6 :
        f=1000*f;
        break;
    case 7 :
        f=0.0005399568034557236*f;
        break;
    case 8 :
        f=1.0936130000578717*f;
        break;
    }
    sprintf(y, "%lf", f);


}
char trans_arie(char x[256],char y[256],int i,int j)
{
    long double f=trans_char_float(x);

    switch (i)
    {
    case 0 :
        f=100*f;
        break;
    case 1 :
        f=4046.856119*f;
        break;
    case 2 :
        f=0.0001*f;
        break;
    case 3 :
        f=0.0929*f;
        break;
    case 4 :
        f=10000*f;
        break;
    case 5 :
        f=0.0006452*f;
        break;
    case 6 :
        f=1000000*f;
        break;
    case 7 :
        f=1*f;
        break;
    case 8 :
        f=0.000001*f;
        break;
    case 9 :
        f=0.836127816*f;
        break;
    }

    switch (j)
    {
    case 0 :
        f=0.01*f;
        break;
    case 1 :
        f=0.00024710539999309525*f;
        break;
    case 2 :
        f=10000*f;
        break;
    case 3 :
        f=10.764262648008613*f;
        break;
    case 4 :
        f=0.0001*f;
        break;
    case 5 :
        f=1549.9070055796653*f;
        break;
    case 6 :
        f=0.000001*f;
        break;
    case 7 :
        f=1*f;
        break;
    case 8 :
        f=1000000*f;
        break;
    case 9 :
        f=1.195989394042597*f;
        break;
    }
    sprintf(y, "%lf", f);


}

char trans_volum(char x[256],char y[256],int i,int j)
{
    long double f=trans_char_float(x);

    switch (i)
    {
    case 0 :
        f=0.00001*f;
        break;
    case 1 :
        f=0.000001*f;
        break;
    case 2 :
        f=0.0001*f;
        break;
    case 3 :
        f=0.001*f;
        break;
    case 4 :
        f=0.02832*f;
        break;
    case 5 :
        f=0.004546*f;
        break;
    case 6 :
        f=0.003785*f;
        break;
    case 7 :
        f=0.000016387064*f;
        break;
    case 8 :
        f=0.001*f;
        break;
    case 9 :
        f=1*f;
        break;
    case 10 :
        f=0.000001*f;
        break;
    case 11 :
        f=0.764554858*f;
        break;
    }

    switch (j)
    {
    case 0 :
        f=100000*f;
        break;
    case 1 :
        f=1000000*f;
        break;
    case 2 :
        f=10000*f;
        break;
    case 3 :
        f=1000*f;
        break;
    case 4 :
        f=35.3146667*f;
        break;
    case 5 :
        f=219.969157*f;
        break;
    case 6 :
        f=264.172052*f;
        break;
    case 7 :
        f=61023.7441*f;
        break;
    case 8 :
        f=1000*f;
        break;
    case 9 :
        f=1*f;
        break;
    case 10:
        f=1000000*f;
        break;
    case 11:
        f=1.30795062*f;
        break;
    }
    sprintf(y, "%lf", f);


}
char trans_timp(char x[256],char y[256],int i,int j)
{
    long double f=trans_char_float(x);

    switch (i)
    {
    case 0 :
        f=f/3600;
        break;
    case 1 :
        f=f/60;
        break;
    case 2 :
        f=1*f;
        break;
    case 3 :
        f=24*f;
        break;
    case 4 :
        f=168*f;
        break;
    case 5 :
        f=5208*f;
        break;
    }

    switch (j)
    {
    case 0 :
        f=3600*f;
        break;
    case 1 :
        f=60*f;
        break;
    case 2 :
        f=1*f;
        break;
    case 3 :
        f=f/24;
        break;
    case 4 :
        f=f/168;
        break;
    case 5 :
        f=f/5208;
        break;
    }
    sprintf(y, "%lf", f);
}
char trans_viteza(char x[256],char y[256],int i,int j)
{
    long double f=trans_char_float(x);

    switch (i)
    {
    case 0 :
        f=0.0001666667*f;
        break;
    case 1 :
        f=0.01*f;
        break;
    case 2 :
        f=0.000084667*f;
        break;
    case 3 :
        f=0.00508*f;
        break;
    case 4 :
        f=0.3048*f;
        break;
    case 5 :
        f=0.0254*f;
        break;
    case 6 :
        f=0.27777778*f;
        break;
    case 7 :
        f=16.666667*f;
        break;
    case 8 :
        f=1000*f;
        break;
    case 9 :
        f=0.000277778*f;
        break;
    case 10 :
        f=0.016666667*f;
        break;
    case 11 :
        f=1*f;
        break;
    case 12 :
        f=1609.344*f;
        break;
    case 13 :
        f=26.8224*f;
        break;
    case 14 :
        f=0.44704*f;
        break;
    case 15 :
        f=0.5144*f;
        break;
    case 16 :
        f=0.000254*f;
        break;
    case 17 :
        f=0.015240004*f;
        break;
    case 18 :
        f=0.9144002494*f;
        break;
    }

    switch (j)
    {
    case 0 :
        f=6000*f;
        break;
    case 1 :
        f=100*f;
        break;
    case 2 :
        f=11810.9771221 *f;
        break;
    case 3 :
        f=196.8503937*f;
        break;
    case 4 :
        f=3.28083*f;
        break;
    case 5 :
        f=39.3700787*f;
        break;
    case 6 :
        f=3.6*f;
        break;
    case 7 :
        f=0.06*f;
        break;
    case 8 :
        f=0.001*f;
        break;
    case 9 :
        f=3600*f;
        break;
    case 10 :
        f=60*f;
        break;
    case 11 :
        f=1*f;
        break;
    case 12 :
        f=0.000621371192*f;
        break;
    case 13 :
        f=0.03728227153424004*f;
        break;
    case 14 :
        f=2.2369362920 *f;
        break;
    case 15 :
        f=1.94384449*f;
        break;
    case 16 :
        f=11810.9771221*f;
        break;
    case 17 :
        f=196.8503937*f;
        break;
    case 18 :
        f=3.28083*f;
        break;
    }
    sprintf(y, "%lf", f);
}

char trans_temperatura(char x[256],char y[256],int i,int j)
{
    long double f=trans_char_float(x);

    switch (i)
    {
    case 0 :
        f=1*f;
        break;
    case 1 :
        f=(f-32)/1.8;
        break;
    case 2 :
        f=f-273.15;
        break;
    case 3 :
        f=(f - 491.67) * 5/9;
        break;
    }

    switch (j)
    {
    case 0 :
        f=1*f;
        break;
    case 1 :
        f=f*1.8 + 32;
        break;
    case 2 :
        f=f+273.15;
        break;
    case 3 :
        f=(f + 273.15) * 1.8;
        break;

    }
    sprintf(y, "%lf", f);
}

char trans_masa(char x[256],char y[256],int i,int j)
{
    long double f=trans_char_float(x);

    switch (i)
    {
    case 0 :
        f=0.001*f;
        break;
    case 1 :
        f=1*f;
        break;
    case 2 :
        f=0.453592*f;
        break;
    case 3 :
        f=1000*f;
        break;
    case 4 :
        f=0.0283495*f;
        break;
    case 5 :
        f=0.0311035*f;
        break;
    case 6 :
        f=14.593903*f;
        break;
    case 7 :
        f=1000*f;
        break;
    }

    switch (j)
    {
    case 0 :
        f=1000*f;
        break;
    case 1 :
        f=1*f;
        break;
    case 2 :
        f=2.20462*f;
        break;
    case 3 :
        f=1000000*f;
        break;
    case 4 :
        f=35.27399072*f;
        break;
    case 5 :
        f=32.15072259*f;
        break;
    case 6 :
        f=0.068521765*f;
        break;
    case 7 :
        f=0.001*f;
        break;
    }
    sprintf(y, "%lf", f);
}

char trans_energie(char x[256],char y[256],int i,int j)
{
    long double f=trans_char_float(x);

    switch (i)
    {
    case 0 :
        f=1055.0551*f;
        break;
    case 1 :
        f=1054.3495*f;
        break;
    case 2 :
        f=4186.7967*f;
        break;
    case 3 :
        f=4.184*f;
        break;
    case 4 :
        f=1*f;
        break;
    case 5 :
        f=1*f;
        break;
    case 6 :
        f=3599.9971*f;
        break;
    }

    switch (j)
    {
    case 0 :
        f=0.0009*f;
        break;
    case 1 :
        f=0.0009*f;
        break;
    case 2 :
        f=0.0002*f;
        break;
    case 3 :
        f=0.239*f;
        break;
    case 4 :
        f=1*f;
        break;
    case 5 :
        f=1*f;
        break;
    case 6 :
        f=0.0003*f;
        break;
    }
    sprintf(y, "%lf", f);
}

char trans_presiune(char x[256],char y[256],int i,int j)
{
    long double f=trans_char_float(x);

    switch (i)
    {
    case 0 :
        f=98066.5*f;
        break;
    case 1 :
        f=101325.2738*f;
        break;
    case 2 :
        f=98000*f;
        break;
    case 3 :
        f=248.84*f;
        break;
    case 4 :
        f=3386.38815789*f;
        break;
    case 5 :
        f=98000*f;
        break;
    case 6 :
        f=1000*f;
        break;
    case 7 :
        f=47.88*f;
        break;
    case 8 :
        f=98*f;
        break;
    case 9 :
        f=9800*f;
        break;
    case 10 :
        f=9800*f;
        break;
    case 11 :
        f=9.8*f;
        break;
    case 12 :
        f=133.322368421*f;
        break;
    case 13 :
        f=1000000*f;
        break;
    case 14 :
        f=1000000*f;
        break;
    case 15 :
        f=1*f;
        break;
    case 16 :
        f=6894.75729317*f;
        break;
    case 17 :
        f=133.322368421*f;
        break;
    }

    switch (j)
    {
    case 0 :
        f=0.000010197162129779282*f;
        break;
    case 1 :
        f=0.000009869205998632101*f;
        break;
    case 2 :
        f=0.000010204081632653061 *f;
        break;
    case 3 :
        f=0.0040186465198521135*f;
        break;
    case 4 :
        f=0.00029529987508079483*f;
        break;
    case 5 :
        f=0.000010204081632653061*f;
        break;
    case 6 :
        f=0.001*f;
        break;
    case 7 :
        f=0.020885547201336674*f;
        break;
    case 8 :
        f=0.01020408163265306*f;
        break;
    case 9 :
        f=0.00010204081632653062*f;
        break;
    case 10 :
        f=0.00010204081632653062*f;
        break;
    case 11 :
        f=0.1020408163265306*f;
        break;
    case 12 :
        f=0.007500616827044659*f;
        break;
    case 13 :
        f=0.000001*f;
        break;
    case 14 :
        f=0.000001 *f;
        break;
    case 15 :
        f=1*f;
        break;
    case 16 :
        f=0.00014503773773017476*f;
        break;
    case 17 :
        f=0.007500616827044659*f;
        break;
    }
    sprintf(y, "%lf", f);
}

char trans_densitate(char x[256],char y[256],int i,int j)
{
    long double f=trans_char_float(x);

    switch (i)
    {
    case 0 :
        f=1000*f;
        break;
    case 1 :
        f=1*f;
        break;
    case 2 :
        f=0.001*f;
        break;
    case 3 :
        f=1000*f;
        break;
    case 4 :
        f=1000000*f;
        break;
    case 5 :
        f=1000000*f;
        break;
    case 6 :
        f=1000*f;
        break;
    case 7 :
        f=1*f;
        break;
    case 8 :
        f=16.017940092904052*f;
        break;
    case 9 :
        f=27679.89858*f;
        break;
    case 10 :
        f=99.77637266*f;
        break;
    case 11 :
        f=119.8264273*f;
        break;
    case 12 :
        f=1*f;
        break;
    case 13 :
        f=0.001*f;
        break;
    case 14 :
        f=0.000001*f;
        break;
    case 15 :
        f=6.236023291*f;
        break;
    case 16 :
        f=7.489150094*f;
        break;
    case 17 :
        f=1.001153961*f;
        break;
    case 18 :
        f=1729.994044*f;
        break;
    case 19 :
        f=1000*f;
        break;
    }

    switch (j)
    {
    case 0 :
        f=0.001*f;
        break;
    case 1 :
        f=1*f;
        break;
    case 2 :
        f=1000 *f;
        break;
    case 3 :
        f=0.001*f;
        break;
    case 4 :
        f=0.000001*f;
        break;
    case 5 :
        f=0.000001*f;
        break;
    case 6 :
        f=0.001*f;
        break;
    case 7 :
        f=1*f;
        break;
    case 8 :
        f=0.06243*f;
        break;
    case 9 :
        f=0.00003612730000111149*f;
        break;
    case 10 :
        f=0.010022412855272063*f;
        break;
    case 11 :
        f=0.00834540445319611*f;
        break;
    case 12 :
        f=1*f;
        break;
    case 13 :
        f=1000*f;
        break;
    case 14 :
        f=1000000 *f;
        break;
    case 15 :
        f=0.16035860569078172*f;
        break;
    case 16 :
        f=0.13352649999646274*f;
        break;
    case 17 :
        f=0.9988473690911163*f;
        break;
    case 18 :
        f=0.0005780366721308782*f;
        break;
    case 19 :
        f=0.001*f;
        break;
    }
    sprintf(y, "%lf", f);
}

char trans_consum(char x[256],char y[256],int i,int j)
{
    long double f=trans_char_float(x);

    switch (i)
    {
    case 0 :
        f=1*f;
        break;
    case 1 :
        f=100/f;
        break;
    case 2 :
        f=235.21/f;
        break;
    case 3 :
        f=2.352*f;
        break;
    case 4 :
        f=62.14/f;
        break;
    case 5 :
        f=0.6214*f;
        break;
    }

    switch (j)
    {
    case 0 :
        f=1*f;
        break;
    case 1 :
        f=100/f;
        break;
    case 2 :
        f=235.21/f;
        break;
    case 3 :
        f=0.4251*f;
        break;
    case 4 :
        f=62.14/f;
        break;
    case 5 :
        f=1.609*f;
        break;
    }
    sprintf(y, "%lf", f);
}
